pkg/webhooks/kafkacluster_validator.go

```diff
@@ -133,3 +157,65 @@ func getMissingMounthPathLocation(mounthPath string, kafkaClusterSpec *banzaiclo
 	}
 	return fromConfigGroup
 }
+
+// checkListeners validates the spec.listenersConfig object
+func checkInternalAndExternalListeners(kafkaClusterSpec *banzaicloudv1beta1.KafkaClusterSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, checkUniqueListenerContainerPort(kafkaClusterSpec.ListenersConfig)...)
+
+	allErrs = append(allErrs, checkExternalListenerStartingPort(kafkaClusterSpec)...)
+
+	return allErrs
+}
+
+// checkUniqueListenerContainerPort checks for duplicate containerPort numbers across both internal and external listeners
+// which would subsequently generate a "Duplicate value" error when creating a Service which accumulates all these ports.
+// The first time a port number is found will not be reported as duplicate; only subsequent instances using that port are.
+// (this is done to keep in tune with the way the K8s Service would report the "Duplicate value" error which ignores the first instance)
+func checkUniqueListenerContainerPort(listeners banzaicloudv1beta1.ListenersConfig) field.ErrorList {
+	var allErrs field.ErrorList
+	var containerPorts = make(map[int32]int)
+
+	for i, intListener := range listeners.InternalListeners {
+		containerPorts[intListener.ContainerPort] += 1
+		if containerPorts[intListener.ContainerPort] > 1 {
+			fldErr := field.Duplicate(field.NewPath("spec").Child("listenersConfig").Child("internalListeners").Index(i).Child("containerPort"), intListener.ContainerPort)
```

Comment:
**Question**: Wouldn't it be better for the user experience to report the listener names of the duplicate ports here? (storing the names in a `map[int32][]string` and when the length is over 1 at the end of the iteration then report all so the user knows what to fix exactly)

Same for the iteration following this one on external listeners

Reply:
For reference, this is what we log in the current interaction of this PR for that part of the code:
```
The KafkaCluster "kafka" is invalid:
* spec.listenersConfig.internalListeners[1].containerPort: Duplicate value: 29095
* spec.listenersConfig.externalListeners[1].containerPort: Duplicate value: 29095
```

I like the idea of having the listener names as part of what we report back (it also fits with what is done in the other validation from this PR).  
I am, however, not clear on what you're suggesting. 

Right now, we iterate through the config and as we find a repeated containerPort, we add a `field.Error` (of the `field.Duplicate()` kind) pointing to that specific `containerPort` field (e.g. spec.listenersConfig.internalListeners[1].containerPort)

I understand how we can get aggregate the info (small change; the map definition you suggested) but I don't understand to what field in the spec we would associate the error to, in this case.

<br></br>
Side note:
It is true that one downside of `field.Duplicate()` is that it doesn't allow us to add any `detail string` like `field.Invalid` does.
```
func field.Duplicate(field *field.Path, value interface{}) *field.Error

func field.Invalid(field *field.Path, value interface{}, detail string) *field.Error
```

Reply:
Oh, I see, I didn't think this through.
The field would be the same, only the details would be different, but I overlooked the fact in this place we cannot define details for the duplicate error so I think you can leave it the way it is, because the listener config index is already encoded in the error message.